### PR TITLE
Allow record/map to transform type with optional properties

### PR DESCRIPTION
**Pull request by baoshan**
The current implementation of `record/map` refuses to map a type with optional properties:
```
import * as R from "./record.ts";
type Foo = { bar: number; baz?: number };
const foo: Foo = { bar: 1 };
R.map((n: number) => n + 1)(foo);
                            ^^^
```

Because:
```
Argument of type 'Foo' is not assignable to parameter of type '{ bar: number; baz: number; }'.
  Property 'baz' is optional in type 'Foo' but required in type '{ bar: number; baz: number; }'
```

This PR allows such usage.

### CODE REVIEW

**Comment by baoshan** [comment]
Any chance this PR could have a review? Thanks.

File: record.ts [hunk 1]
```diff
@@ -22,15 +22,25 @@ const compareStrings = toCompare(ordString);
 
 const sortStrings = (keys: string[]): string[] => keys.sort(compareStrings);
 
+/**
+ * Creates a new object with the same keys of `ta`. Values are transformed
+ * using `fai`.
+ *
+ * @example
```
Comment (pixeleet) [inline record.ts:29]: nitpick (non-blocking): 
@baoshan we don't currently use the @example tag anywhere for examples, they're simply fenced in a code block with `ts` flag

@baetheus we should decide for one style, either using @example tag for all examples, or none

That is, my work on #47 currently depends on the fact that it's a code fence block and I'm not parsing tagged blocks. By no means it's fixed as I only have some POC code atm, regardless, I think we should be consistent about the way we produce examples.

**Comment by pixeleet** [comment]
@baoshan Thank you for your contribution!
Apologies for the non-timely review, we're in the cucumber season.

There's one non-blocking nitpick that we should discuss with @baetheus, but otherwise, LGTM.

File: record.ts (hunk 1 above)
Comment (pixeleet) [reply record.ts:29]: Resolved: as it's non blocking, this is more for discussion than blocking the PR.

**Review by pixeleet — APPROVED** [review]
LGTM

pending 1 non-blocking discussion/nitpick.

**Comment by baoshan** [comment]
Thanks for the review.

Switched to fenced ts block for now. Feel free to let me know if you want some hands to port existing examples to any chosen style.

File: record.ts (hunk 1 above)
Comment (baetheus) [reply record.ts]: I don't think there is any syntax parsing in deno for the ```@example``` tag. However, I don't think it matters either way. For consistency, let's start using @example even though it isn't specified in https://deno.land/manual@v1.24.3/testing/documentation#documentation-tests 

It's not necessary to make a repo-wide change unless either of you feel so inclined. We can be inconsistent here until someone with more desire for perfect styling than I comes along with a PR.

**Review by baetheus — APPROVED** [review]

**Comment by baetheus** [comment]
This looks good, just needs another deno fmt and I'll merge it right in.

**Pull request merged**